src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ServerCertificates.cs

```diff
@@ -211,6 +212,54 @@ public static IEnumerable<object[]> UseCallback_ValidCertificate_ExpectedValuesD
             }
         }
 
+        [Fact]
+        public async Task PostAsync_Post_ChannelBinding_ConfiguredCorrectly()
+        {
+            var content = new ChannelBindingAwareContent("Test contest");
+            using (var client = new HttpClient())
+            using (HttpResponseMessage response = await client.PostAsync(HttpTestServers.SecureRemoteEchoServer, content))
+            {
+                // Validate status.
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+                // Validate the ChannelBinding object exists.
+                ChannelBinding channelBinding = content.ChannelBinding;
+                Assert.NotNull(channelBinding);
+
+                // Validate the ChannelBinding's validity.
+                if (BackendSupportsCustomCertificateHandling)
+                {
+                    Assert.False(channelBinding.IsInvalid, "Expected valid binding");
+                    Assert.NotEqual(IntPtr.Zero, channelBinding.DangerousGetHandle());
+
+                    // Validate the ChannelBinding's description.
+                    string channelBindingDescription = channelBinding.ToString();
+                    Assert.NotNull(channelBindingDescription);
+                    Assert.NotEmpty(channelBindingDescription);
+                    Assert.True((channelBindingDescription.Length + 1) % 3 == 0, $"Unexpected length {channelBindingDescription.Length}");
+                    for (int i = 0; i < channelBindingDescription.Length; i++)
+                    {
+                        char c = channelBindingDescription[i];
+                        if (i % 3 == 2)
+                        {
+                            Assert.Equal(' ', c);
+                        }
+                        else
+                        {
+                            Assert.True((c >= '0' && c <= '9') || (c >= 'A' && c <= 'F'), $"Expected hex, got {c}");
```

Comment:
The implementation will always produce upper-case.  There are no guarantees in the docs that these will be upper-case... there are no guarantees in the docs that any such value will come from ToString.